presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaSplitManager.java

```diff
@@ -100,7 +99,8 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
                 SimpleConsumer leaderConsumer = consumerManager.getConsumer(partitionLeader);
                 // Kafka contains a reverse list of "end - start" pairs for the splits
 
-                List<HostAddress> partitionNodes = ImmutableList.copyOf(Lists.transform(part.isr(), KafkaSplitManager::brokerToHostAddress));
+                // Always talk to partition leader
+                List<HostAddress> partitionNodes = ImmutableList.of(partitionLeader);
```

Comment:
Why do we need a list of nodes here? We just need a single `HostAddress`. Similarly `KafkaSplit` doesn't need to have a list of nodes anymore, a single `HostAddress` that points to the leader should do it. If we update `KafkaSplit` then in `KafkaRecordSet::openFetchRequest()` we can just get the leader instead of getting the `0th` element of the node list

```
SimpleConsumer consumer = consumerManager.getConsumer(split.getNodes().get(0));
```